packages/core/src/utils/deal-score.ts

```diff
@@ -0,0 +1,95 @@
+import {calculateDiscount} from './discount'
+
+import {getCheapestRate} from './cheapest-rate'
+
+import {HotelId, HotelOfferEntity, Hotel} from '../types'
+
+export type DealScores = Record<HotelId, number>
+
+type DealScoreInput = [
+  HotelId,
+  number | undefined,
+  number | undefined,
+  number | undefined
+]
+
+/**
+ * Calculates deal scores from extracted parameters
+ *
+ * @param input - DealScoreInput[]
+ *
+ * @returns DealScores
+ */
+export function calculateDealScores(input: DealScoreInput[] = []): DealScores {
+  const discounts: Record<HotelId, number> = {}
+  const dealScores: DealScores = {}
+
+  input.forEach((item) => {
+    const [hotelId, cheapestRate, regularFrom, regularTo] = item
```

Comment:
```suggestion
    const [hotelId, cheapestRate, historicalLowerBound, historicalUpperBound] = item
```

I prefer lower / upper bound because the from / to implicates a "discount" in some way, like the "from" price.